### PR TITLE
feat: d2l-navigation-link-icon component

### PR DESCRIPTION
**Pull request by dlockhart**
This introduces a `<d2l-navigation-link-icon>` component, which is a specialized version of `<d2l-navigation-link>` that comes with an icon and text built in. The text can optionally be visually hidden.

After this, I have several other refactoring tasks to do, but wanted to try and keep this as clean as possible:
- Switch `<d2l-navigation-link-back>` to use this
- Introduce a corresponding `<d2l-navigation-button-icon>`
- Switch `<d2l-navigation-button-close>` to use that
- Since there are only a couple actual uses of `<d2l-navigation-link>`/`<d2l-navigation-button>` and only a single usage of `<d2l-navigation-link-image>` in the LMS, I'd like to try and switch them over to these new components and kill them off
- Clean up how translations are done in this repo... there's a Gulpfile and... yeah it's weird

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -151,6 +151,23 @@ Add the `d2l-navigation-main-footer` component, and provide elements for the `ma
 
 ---
 
+### d2l-navigation-link-icon
+
+Similar to `<d2l-navigation-button-icon>`, a link that comes with an icon and optional text.
```
Comment (dlockhart) [inline README.md:156]: Getting a bit ahead since that component won't exist until the next PR.

File: d2l-navigation-link.js [hunk 2]
```diff
@@ -57,7 +57,7 @@ class D2LNavigationLink extends mixinBehaviors([D2L.PolymerBehaviors.FocusableBe
 					display: none;
 				}
 			</style>
-			<a href$="[[href]]" title$="[[text]]">
+			<a class="d2l-focusable" href$="[[href]]" title$="[[text]]">
```
Comment (dlockhart) [inline d2l-navigation-link.js:60]: I incorrectly removed this #143, but it was needed by the `FocusableBehavior` to delegate focus to the anchor.

File: d2l-navigation-styles.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+		text-decoration: none;
+		vertical-align: middle;
+	}
+	a:hover,
+	a:focus {
+		color: var(--d2l-color-celestine);
+		--d2l-icon-fill-color: var(--d2l-color-celestine);
+		outline: none;
+	}
+`;
```
Comment (dlockhart) [inline d2l-navigation-styles.js:40]: These are duplicates of the Polymer style mixins, but those will go away shortly!

File: test/button.visual-diff.html [hunk 4]
```diff
@@ -11,41 +11,25 @@
 			import '../../d2l-navigation-button.js';
 			import '../../d2l-navigation-button-close.js';
 			import '../../d2l-navigation-button-notification-icon.js';
-			import '../../d2l-navigation-link.js';
-			import '../../d2l-navigation-link-back.js';
-			import '../../d2l-navigation-link-image.js';
```
Comment (dlockhart) [inline test/button.visual-diff.html:16]: I split the link visual-diffs out from the button ones as the filenames were getting insane.

File: d2l-navigation-link-icon.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+ */
+class NavigationLinkIcon extends FocusMixin(LitElement) {
+
+	static get properties() {
+		return {
+			/**
+			 * REQUIRED: URL or URL fragment of the link
+			 * @type {string}
+			 */
+			href: {
```
Comment (svanherk) [inline d2l-navigation-link-icon.js:20]: _SUPER_ minor, but I like the one-line style we use in `core` for `properties`.  This spread-out style makes me think of Polymer days lol

**Review by svanherk — APPROVED** [review]

File: d2l-navigation-link-icon.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+		super();
+		this.textHidden = false;
+	}
+
+	static get focusElementSelector() {
+		return 'a';
+	}
+
+	render() {
+		const highlight = this.href ? html`<span class="d2l-navigation-highlight-border"></span>` : nothing;
```
[the remark this answers is not in the document]
Comment (dlockhart) [reply d2l-navigation-link-icon.js:62]: Potentially none, and I'm open to removing this. Mostly I want to be able to completely replace `<d2l-navigation-link>` and for whatever reason it does this. Unfortunately [activity feed](https://github.com/Brightspace/d2l-class-stream-ui/blob/ca9c03e34e25fdce1f818984e52ec8a057fb1b82/components/d2l-cs-planned-activities-home.js#L105) is relying on this weird quirk as well, but I _think_ I can likely convince them to use something else.

I'm leaning towards removing it.

File: demo/button-link.html [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+						<d2l-icon icon="tier2:gear" style="margin-right: 10px;"></d2l-icon>
+						Settings
+					</d2l-navigation-link>
+				</div>
+			</d2l-demo-snippet>
+
+			<h2>d2l-navigation-link-icon</h2>
+			<d2l-demo-snippet>
+				<div class="wrapper">
+					<d2l-navigation-link-icon icon="tier3:gear" text="Settings"></d2l-navigation-link-icon>
```
Comment (svanherk) [inline demo/button-link.html:50]: Did you want to remove the `href` here, or was that just for testing the timeout?

File: d2l-navigation-link-icon.js [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+		return html`
+			<a href="${ifDefined(this.href)}" title="${ifDefined(this.textHidden ? this.text : undefined)}">
+				<span class="d2l-navigation-highlight-border"></span>
+				<d2l-icon icon="${this.icon}"></d2l-icon>
+				<span class="${classMap(textClasses)}">${this.text}</span>
+			</a>
+		`;
+	}
+
+	async updated(changedProperties) {
```
[the remark this answers is not in the document]
Comment (dlockhart) [reply d2l-navigation-link-icon.js]: Good catch -- I had copied this exception code from `LabelledMixin` and it was more complex. Fixed.

**Pull request merged**